sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala

```diff
@@ -48,6 +48,13 @@ case class AQEShuffleReadExec private(
     assert(partitionSpecs.forall(_.isInstanceOf[PartialMapperPartitionSpec]))
   }
 
+  private def shuffleStage = child match {
+    case stage: ShuffleQueryStageExec => Some(stage)
+    case _ => None
+  }
+
+  override protected def isCanonicalizedPlan: Boolean = shuffleStage.isEmpty
```

Comment:
what's wrong with the default implementation of `isCanonicalizedPlan`?

Reply:
The default implementation of isCanonicalizedPlan is correct. here we just make it clear: if the plan is cannonicalized then the shuffle stage is empty. I change to use `super.isCanonicalizedPlan && shuffleStage.isEmpty`